CDC-Data-Reconciliation-Backend/compare.py

```diff
@@ -106,6 +133,15 @@ def comp(state_dict, cdc_dict):
         cdc_row = cdc_dict[cdc_case_id]
         results.append(CaseResult(cdc_case_id, cdc_row['EventCode'],
                        cdc_row['MMWRYear'], cdc_row['MMWRWeek'], "Case ID not found in State CSV File", "4"))
+        
+        # adding in missing from state count, total case count, and caseID to the stats dict
+        # only counting cases that are not duplicates, otherwise counting as duplicate
+        if cdc_row['CaseID'] not in stats[cdc_row['EventCode']]['caseIDs']:
+            stats[cdc_row['EventCode']]['totalMissingState'] += 1
+            stats[cdc_row['EventCode']]['totalCases'] += 1
+            stats[cdc_row['EventCode']]['caseIDs'].append(cdc_row['CaseID'])
+        else:
+            stats[cdc_row['EventCode']]['totalDuplicates'] += 1
```

Comment:
First, you are not guaranteed that the EventCode already exists within the stats dictionary, and thus you should add an entry if it is not. Also, you are guaranteed here that the CaseID will not be present within the state as this for loop is going over all caseID's that were not in the state data.

'TotalDuplicates' should be calculated in the get_cdc_dict function as we disregard duplicates from CDC there and also report if there is a CDC duplicate.